### PR TITLE
Implementation of vending machine by coins and credit card

### PR DESCRIPTION
**Pull request by yodanlevy**

### CODE REVIEW

File: VendingMachine/VendingMachine/CreditCard.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public class CreditCard
+    {
+        public string Number;
+
+        public CreditCard(string number)
+        {
+            this.Number = number;
+        }
+
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/CreditCard.cs]: Delete empty line

File: VendingMachine/VendingMachine/CreditCard.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿namespace VendingMachine
+{
+    public class CreditCard
+    {
+        public string Number;
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/CreditCard.cs]: Why public?
Pay attention that you'll always wish to encapsulate your properties as much as possible.
 Here you can set the field to private and implement a getter to it

File: VendingMachine/VendingMachine/DataInitializer.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace VendingMachine
+{
+    public class DataInitializer
+    {
+        public List<Drink> InitializeDrink()
+        {
+            Drink CocaCola= new Drink("Coca-Cola", "A1", 5);
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/DataInitializer.cs]: Why drink gets "A1" as an argument ?
The drink should not know where it is placed.
You could implement it with a dictionary where "A1" for example is a key there and the value is the drink it self

File: VendingMachine/VendingMachine/DataInitializer.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            Drink CocaCola= new Drink("Coca-Cola", "A1", 5);
+            Drink CocaColaZero = new Drink("Coca-Cola Zero", "A2", 5);
+            Drink Sprite = new Drink("Sprite", "A3", 5);
+            Drink SpriteZero = new Drink("Sprite Zero", "A4", 5);
+            Drink OrangeJuice = new Drink("Orange Juice", "B1", 6);
+            Drink GrapeJuice = new Drink("Grape Juice", "B2", 6);
+            Drink Water = new Drink("Water", "B3", 4);
+
+            List<Drink> drinkMenu = new List<Drink>();
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/DataInitializer.cs]: If this is a list of Drink type, 
why is it a menu?
Rename to just `drinks`

File: VendingMachine/VendingMachine/Drink.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿namespace VendingMachine
+{
+    public class Drink
+    {
+        public string DrinkName;
+        public string SerialNumber;
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Drink.cs]: Why `SerialNumber` is a string?

File: VendingMachine/VendingMachine.sln
Comment (dar-levy) [inline VendingMachine/VendingMachine.sln:1]: What is this file?
Search the web and comment below what this file is for.

File: VendingMachine/VendingMachine/DataInitializer.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace VendingMachine
+{
+    public class DataInitializer
+    {
+        public List<Drink> InitializeDrink()
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/DataInitializer.cs]: Do you initialize just one drink here?

File: VendingMachine/VendingMachine/DataInitializer.cs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            drinkMenu.Add(Sprite);
+            drinkMenu.Add(SpriteZero);
+            drinkMenu.Add(OrangeJuice);
+            drinkMenu.Add(GrapeJuice);
+            drinkMenu.Add(Water);
+
+            return drinkMenu;
+        }
+
+        public List<CreditCard> InitializeCreditCard()
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/DataInitializer.cs]: Do you initialize just one credit card here?

File: VendingMachine/VendingMachine/Drink.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        public double Price;
+
+        public Drink(string drinkName, string serialNumber, double price)
+        {
+            this.DrinkName = drinkName;
+            this.SerialNumber = serialNumber;
+            this.Price = price;
+        }
+
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Drink.cs]: Delete empty lines

File: VendingMachine/VendingMachine/IO.cs [hunk 9]
```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace VendingMachine
+{
+    public class IO
+    {
+        public static DataInitializer dI = new DataInitializer();
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: What the hell is `dI` ???

Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: Why is it static ????

File: VendingMachine/VendingMachine/IO.cs [hunk 10]
```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace VendingMachine
+{
+    public class IO
+    {
+        public static DataInitializer dI = new DataInitializer();
+        public List<Drink> _drinkList = dI.InitializeDrink();
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: What if tommorow you would like to change this collection to a dictionary?
Just `_drinks` here

Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: Why is it public and the naming convention is of a private field?

File: VendingMachine/VendingMachine/IO.cs [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+        public static DataInitializer dI = new DataInitializer();
+        public List<Drink> _drinkList = dI.InitializeDrink();
+        public List<CreditCard> ValidCards = dI.InitializeCreditCard();
+
+        public void PrintDrinks()
+        {
+            Console.WriteLine("Please choose a drink:");
+            foreach (var VARIABLE in _drinkList)
+            {
+                Console.WriteLine(VARIABLE.SerialNumber + ". " + VARIABLE.DrinkName + " = " + VARIABLE.Price + " NIS" + "\n");
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: Check how to use `$` sign to print without these `+`

File: VendingMachine/VendingMachine/IO.cs [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+            }
+        }
+
+        public int ChoosePaymentMethod()
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: This function is too big, 
extract some code lines to another private function

File: VendingMachine/VendingMachine/IO.cs [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+        {
+            Console.WriteLine("For payment in cash please press 1 \n" +
+                              "For payment with credit card please press 2");
+
+            int paymentMethod = 0;
+            string userPaymentMethod = Console.ReadLine();
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs:31]: nice variable name!

File: VendingMachine/VendingMachine/IO.cs [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+
+            while (!string.IsNullOrWhiteSpace(userPaymentMethod))
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: Why do you check this?
Are you sure you need to check this condition?

File: VendingMachine/VendingMachine/IO.cs [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+                if (Validator.IsPaymentMethodValid(userPaymentMethod))
+                {
+                    paymentMethod = int.Parse(userPaymentMethod);
+                    return paymentMethod;
+                }
+
+                Console.WriteLine("Payment method is not valid \n" +
+                                  "Please try again");
+                userPaymentMethod = Console.ReadLine();
+
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: Empty line

File: VendingMachine/VendingMachine/IO.cs [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+                Console.WriteLine(VARIABLE.SerialNumber + ". " + VARIABLE.DrinkName + " = " + VARIABLE.Price + " NIS" + "\n");
+            }
+        }
+
+        public int ChoosePaymentMethod()
+        {
+            Console.WriteLine("For payment in cash please press 1 \n" +
+                              "For payment with credit card please press 2");
+
+            int paymentMethod = 0;
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: Why do you have this variable?

File: VendingMachine/VendingMachine/IO.cs [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+            string userPaymentMethod = Console.ReadLine();
+
+            while (!string.IsNullOrWhiteSpace(userPaymentMethod))
+            {
+                if (Validator.IsPaymentMethodValid(userPaymentMethod))
+                {
+                    paymentMethod = int.Parse(userPaymentMethod);
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: Simply `return int.Parse(userPaymentMethod)`

File: VendingMachine/VendingMachine/IO.cs (hunk 14 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: Why not using some unique key to exit?
Meaning think of a friendly way to tell the user how to exit your system of return to the main menu

File: VendingMachine/VendingMachine/Validator.cs [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public static bool IsCreditCardValid(string creditCardNumber, List<CreditCard> validCards)
+        {
+            foreach (var VARIABLE in validCards)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Validator.cs]: There is a much better implementation for this foreach loop.
There is a dedicated function for this situation.
Try look for it and reimplement it.
It should take you one line of code

File: VendingMachine/VendingMachine/Validator.cs [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+            {
+                return _coins.Contains(number);
+            }
+
+            return false;
+        }
+
+        public static bool IsPaymentMethodValid(string userPaymentMethod)
+        {
+            if (userPaymentMethod == "1" || userPaymentMethod == "2")
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Validator.cs]: Why not just `return (userPaymentMethod == "1" || userPaymentMethod == "2")` ?

File: VendingMachine/VendingMachine/Validator.cs [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+using System.ComponentModel.Design;
+using System.Linq;
+
+namespace VendingMachine
+{
+    public class Validator
+    {
+        static double[] _coins = { 0.1, 0.5, 1, 2, 5, 10 };
+
+        public static bool IsValid(String str)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Validator.cs]: What object does this method check validity for?
Is it a coin, a credit card, etc...

File: VendingMachine/VendingMachine/Validator.cs [hunk 21]
```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Design;
+using System.Linq;
+
+namespace VendingMachine
+{
+    public class Validator
+    {
+        static double[] _coins = { 0.1, 0.5, 1, 2, 5, 10 };
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Validator.cs]: You should extract this property to a constants file.
Read about Constants file in web and implement it in the project

File: VendingMachine/VendingMachine/Validator.cs (hunk 18 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/Validator.cs]: Read about Lambda expressions as well in c#

File: VendingMachine/VendingMachine/VendingMachine.csproj
Comment (dar-levy) [inline VendingMachine/VendingMachine/VendingMachine.csproj:1]: What is this file for?
Read about it in the web and explain in a comment below

File: VendingMachine/VendingMachine/Validator.cs [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace VendingMachine
+{
+    public class Validator
+    {
+        static double[] _coins = { 0.1, 0.5, 1, 2, 5, 10 };
+
+        public static bool IsValid(String str)
+        {
+            if (double.TryParse(str, out var number))
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Validator.cs]: You can replace this function implementation with just one line.
Think how and reimplement

File: VendingMachine/VendingMachine/Program.cs
Comment (dar-levy) [inline VendingMachine/VendingMachine/Program.cs:1]: BEAUTIFUL Program.cs file 💯

File: VendingMachine/VendingMachine/Manager.cs [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+using System.Text;
+using System.Threading;
+
+namespace VendingMachine
+{
+    class Manager
+    {
+        public IO io = new IO();
+        public double Total = 0.0;
+        public void Start()
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Manager.cs:12]: This start function is too big, extract the conditions inner code to private functions

File: VendingMachine/VendingMachine/Manager.cs [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            io.PrintDrinks();
+
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Manager.cs]: Why this empty line?
DELETE

File: VendingMachine/VendingMachine/Manager.cs [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+            io.PrintDrinks();
+            
+            int paymentMethod = io.ChoosePaymentMethod();
+
+            if (paymentMethod == 1)
+            {
+                Console.WriteLine("Insert coins");
+                string userCoins = Console.ReadLine();
+
+                Total = io.InsertCoins(userCoins, Total);
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Manager.cs]: Why are you sending `Total` as an argument here?
Think of a better way to implement this

File: VendingMachine/VendingMachine/Manager.cs [hunk 26]
```diff
@@ -0,0 +1,36 @@
...
+
+                Drink userDrinkChoice = io.ChooseDrink();
+
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/Manager.cs:34]: DELETE empty line here

File: VendingMachine/VendingMachine/IO.cs [hunk 27]
```diff
@@ -0,0 +1,145 @@
...
+                }
+                else
+                {
+                    Console.WriteLine($"Enjoy your drink! {userDrinkChoice.DrinkName}");
+                    break;
+                }
+            }
+        }
+
+        public double InsertCoins(string coins, double total)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: Why does the function `InsertCoins` get coins as an argument?
from the function's name, one may deduce that the function prints to the user, "Please enter coins",
gets the coins, and returns them.
Find a better name for this function, and implement it by that name.

File: VendingMachine/VendingMachine/IO.cs [hunk 28]
```diff
@@ -0,0 +1,145 @@
...
+            {
+                if (!Validator.IsCreditCardValid(userCreditCard, ValidCards))
+                {
+                    Console.WriteLine("\nYour credit card number is not valid.\n" +
+                                      "Please try again!");
+                    userCreditCard = Console.ReadLine();
+                }
+                else
+                {
+                    Console.WriteLine($"Enjoy your drink! {userDrinkChoice.DrinkName}");
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: This is a duplicated code.
Extract it to another IO function and call once in the Manager as we agreed.

File: VendingMachine/VendingMachine/IO.cs [hunk 29]
```diff
@@ -0,0 +1,145 @@
...
+            Console.WriteLine("Total: " + total);
+            return total;
+        }
+
+        public Drink ChooseDrink()
+        {
+            Console.WriteLine("Choose a drink");
+            string userDrinkChoice = Console.ReadLine();
+            
+            while (SerialNumberValidator(userDrinkChoice) == false)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: What kind of function's name is it?
We said that a function should be named a VERB.
Meaning `ValidateSerialNumber`..

Moreover why does this function gets an argument called `userDrinkChoise` and not a `serialNumber`?

File: VendingMachine/VendingMachine/IO.cs [hunk 30]
```diff
@@ -0,0 +1,145 @@
...
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public Drink IdentifyDrink(string serialNumber)
+        {
+
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: Empty line

File: VendingMachine/VendingMachine/IO.cs [hunk 31]
```diff
@@ -0,0 +1,145 @@
...
+            foreach (var VARIABLE in _drinkList)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: There is a much better way to implement this foreach block code.
There's a dedicated function for that...

Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: Why VARIABLE is all capitalized???????
and why not `foreach (var drink in _drinks)` ????????

File: VendingMachine/VendingMachine/IO.cs [hunk 32]
```diff
@@ -0,0 +1,145 @@
...
+            while (SerialNumberValidator(userDrinkChoice) == false)
+            {
+                Console.WriteLine("This drink serial number is not valid");
+                userDrinkChoice = Console.ReadLine();
+            }
+
+            return IdentifyDrink(userDrinkChoice);
+        }
+
+        public void CheckTotal(double total, Drink usersDrinkChoice)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: Can you think of a better name here?

File: VendingMachine/VendingMachine/IO.cs [hunk 33]
```diff
@@ -0,0 +1,145 @@
...
+            }
+            else
+            {
+                Console.WriteLine("You're missing " + (usersDrinkChoice.Price - total) + " NIS");
+                var userCoins = Console.ReadLine();
+                InsertCoins(userCoins, total);
+            }
+        }
+
+        public bool SerialNumberValidator(string serialNumber)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: NAMING VERB

File: VendingMachine/VendingMachine/IO.cs [hunk 34]
```diff
@@ -0,0 +1,145 @@
...
+        {
+            foreach (var variable in _drinkList)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: There is a dedicated function to implement that.
Search the web for it and reimplement

File: VendingMachine/VendingMachine/CreditCard.cs (hunk 1 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/CreditCard.cs]: fixed

File: VendingMachine/VendingMachine/CreditCard.cs (hunk 2 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/CreditCard.cs]: fixed

File: VendingMachine/VendingMachine/DataInitializer.cs (hunk 3 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/DataInitializer.cs]: Drinks converted to dictionary

File: VendingMachine/VendingMachine/DataInitializer.cs (hunk 4 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/DataInitializer.cs]: Renamed

Comment (yodanlevy) [reply VendingMachine/VendingMachine/DataInitializer.cs]: No

File: VendingMachine/VendingMachine/Drink.cs (hunk 5 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Drink.cs]: Because it contains both letters and numbers

File: VendingMachine/VendingMachine/IO.cs (hunk 30 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Fixed

File: VendingMachine/VendingMachine/IO.cs (hunk 31 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Fixed

File: VendingMachine/VendingMachine/IO.cs (hunk 33 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Fixed

File: VendingMachine/VendingMachine/IO.cs [hunk 35]
```diff
@@ -0,0 +1,144 @@
...
+            {
+                Console.WriteLine("You're missing " + (usersDrinkChoice.Price - total) + " NIS");
+                var userCoins = Console.ReadLine();
+                InsertCoins(userCoins, total);
+            }
+        }
+
+        public bool ValidateSerialNumber(string serialNumber)
+        {
+            foreach (var variable in Drinks)
```
Comment (dar-levy) [inline VendingMachine/VendingMachine/IO.cs]: `return myList.Any(t => t.Message == "hello")`

File: VendingMachine/VendingMachine/IO.cs (hunk 34 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Fixed

File: VendingMachine/VendingMachine/IO.cs (hunk 13 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs:31]: 👻

File: VendingMachine/VendingMachine/IO.cs (hunk 12 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Shortened it without private function

File: VendingMachine/VendingMachine/IO.cs (hunk 16 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Variable removed

File: VendingMachine/VendingMachine/IO.cs (hunk 14 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Done

File: VendingMachine/VendingMachine/IO.cs (hunk 17 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Converted paymentMethod in general to string

File: VendingMachine/VendingMachine/IO.cs (hunk 15 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Removed

File: VendingMachine/VendingMachine/IO.cs (hunk 27 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Renamed to PayInCash

File: VendingMachine/VendingMachine/Manager.cs (hunk 24 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Manager.cs]: Removed

File: VendingMachine/VendingMachine/Manager.cs (hunk 26 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Manager.cs:34]: Removed

File: VendingMachine/VendingMachine/Validator.cs (hunk 20 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Validator.cs]: Renamed to IsCoinValid

File: VendingMachine/VendingMachine/Validator.cs (hunk 22 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Validator.cs]: Fixed

File: VendingMachine/VendingMachine/Validator.cs (hunk 19 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Validator.cs]: Fixed

File: VendingMachine/VendingMachine/Manager.cs (hunk 23 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Manager.cs:12]: Fixed

File: VendingMachine/VendingMachine/Validator.cs (hunk 21 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Validator.cs]: Fixed

File: VendingMachine/VendingMachine/DataInitializer.cs (hunk 6 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/DataInitializer.cs]: Renamed to InitializeDrinks

File: VendingMachine/VendingMachine/Validator.cs (hunk 18 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Validator.cs]: Fixed

File: VendingMachine/VendingMachine/DataInitializer.cs (hunk 7 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/DataInitializer.cs]: Renamed to InitializeCreditCards

File: VendingMachine/VendingMachine/IO.cs (hunk 32 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Renamed

File: VendingMachine/VendingMachine/Manager.cs (hunk 25 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Manager.cs]: Fixed (?)

File: VendingMachine/VendingMachine/Program.cs
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Program.cs:1]: Sensational









Perfection

File: VendingMachine/VendingMachine/VendingMachine.csproj
Comment (yodanlevy) [reply VendingMachine/VendingMachine/VendingMachine.csproj:1]: It's responsible for compiling, packing, and publishing code

File: VendingMachine/VendingMachine.sln
Comment (yodanlevy) [reply VendingMachine/VendingMachine.sln:1]: A structure for organizing projects in Visual Studio in 2 files:
1) text based
2) Binary

Comment (dar-levy) [reply VendingMachine/VendingMachine.sln:1]: ?

File: VendingMachine/VendingMachine/IO.cs (hunk 9 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: ?

File: VendingMachine/VendingMachine/IO.cs (hunk 11 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: ?

File: VendingMachine/VendingMachine/IO.cs (hunk 28 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: ?

File: VendingMachine/VendingMachine/IO.cs (hunk 29 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: ?

File: VendingMachine/VendingMachine/IO.cs (hunk 30 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: I still see empty line

Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: ?

File: VendingMachine/VendingMachine/IO.cs (hunk 31 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/IO.cs]: ?

File: VendingMachine/VendingMachine/Manager.cs (hunk 25 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/Manager.cs]: ?

File: VendingMachine/VendingMachine/Manager.cs (hunk 26 above)
Comment (dar-levy) [reply VendingMachine/VendingMachine/Manager.cs:34]: ?

File: VendingMachine/VendingMachine/VendingMachine.csproj
Comment (dar-levy) [reply VendingMachine/VendingMachine/VendingMachine.csproj:1]: ?

File: VendingMachine/VendingMachine/IO.cs (hunk 35 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Drinks was changed to dictionary, and this line was changed to:
return Drinks.ContainsKey(serialNumber);

File: VendingMachine/VendingMachine/IO.cs (hunk 30 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Fixed

File: VendingMachine/VendingMachine/Manager.cs (hunk 25 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Manager.cs]: Fixed.
It's not updated for you

File: VendingMachine/VendingMachine/Manager.cs (hunk 26 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/Manager.cs:34]: Was already removed.
It's not updated for you

File: VendingMachine/VendingMachine/IO.cs (hunk 29 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Already fixed.
It;s not updated for you

File: VendingMachine/VendingMachine/IO.cs (hunk 11 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Fixed

File: VendingMachine/VendingMachine/IO.cs (hunk 29 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Argument renamed to userSerialNumberChoice

File: VendingMachine/VendingMachine/IO.cs (hunk 31 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: variable renamed to drink

File: VendingMachine/VendingMachine/IO.cs (hunk 9 above)
Comment (yodanlevy) [reply VendingMachine/VendingMachine/IO.cs]: Fixed

**Pull request merged**